homework/home1.1/src/Main.java

```diff
@@ -5,47 +5,57 @@ public class Main {
     public static void main(String[] args) {
         Scanner scanner = new Scanner(System.in);
 
-        // Input coefficients
-        System.out.print("Enter coefficient a: ");
-        while (!scanner.hasNextDouble()) {
-            System.out.print("Invalid input. Enter coefficient a: ");
-            scanner.next();
-        }
-        double a = scanner.nextDouble();
+        double[] coefficients = readCoefficients(scanner);
+        scanner.close();
 
-        if (a == 0) {
-            System.out.println("Coefficient 'a' must not be zero, as this would not be a quadratic equation.");
+        if (coefficients == null) {
+            System.out.println("Invalid input. Please enter valid coefficients.");
             return;
         }
 
-        System.out.print("Enter coefficient b: ");
-        while (!scanner.hasNextDouble()) {
-            System.out.print("Invalid input. Enter coefficient b: ");
-            scanner.next();
+        double a = coefficients[0];
+        double b = coefficients[1];
+        double c = coefficients[2];
+
+        double[] roots = calculateRoots(a, b, c);
+
+        if (roots.length == 0) {
+            System.out.println("The equation does not have real roots.");
+        } else if (roots.length == 1) {
+            System.out.println("The equation has one real root: root = " + roots[0]);
+        } else {
+            System.out.println("The equation has two real roots: root1 = " + roots[0] + ", root2 = " + roots[1]);
         }
-        double b = scanner.nextDouble();
+    }
 
-        System.out.print("Enter coefficient c: ");
-        while (!scanner.hasNextDouble()) {
-            System.out.print("Invalid input. Enter coefficient c: ");
-            scanner.next();
+    private static double[] readCoefficients(Scanner scanner) {
+        double[] coefficients = new double[3];
+
+        for (int i = 0; i < 3; i++) {
```

Comment:
minor: you can use named constant for 3 or length of coefficients container to avoid duplication of hardcoded values